### PR TITLE
Pyproject.toml

### PR DESCRIPTION
**Pull request by KOLANICH**
With your PR, here is a check list:

- [ ] Has test cases written? - not needed
- [x] Has all code lines tested? - no new features introduced
- [x] Has `make format` been run? - yes, but it had introduced irrelevant changes. Reverted every change not touching the lines I have touched.
- [x] Please update CHANGELOG.yml (not CHANGELOG.rst)
- [ ] Has fair amount of documentation if your change is complex - not needed
- [x] Agree on NEW BSD License for your contribution

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -1,3 +1,6 @@
+# generated version file
+pyexcel/__version__.py
```
Comment (chfw) [inline .gitignore:2]: if __version__.py is not seen by git, would it work?

File: pyexcel/__init__.py [hunk 2]
```diff
@@ -46,5 +46,7 @@
     load_from_records,
     load_book_from_memory,
 )
-from .__version__ import __author__, __version__
+from .__version__ import version as __version__
```
Comment (chfw) [inline pyexcel/__init__.py:49]: note here, '__version__.py' won't be seen any more.

**Comment by chfw** [comment]
Your PR is welcome. I will accept it.


And after accepting your PR, the next step from me is to migrate your changes to pypi-mobans project, which serves the top level template for the project meta file.

**Comment by KOLANICH** [comment]
>`setuptools_scm` fetches the version from the tag automatically.

And generates the needed file, it is configured via `write_to` in `tool.setuptools_scm` section.

**Comment by KOLANICH** [comment]
Also, I have done the conversion `setup.cfg` `->` `pyproject.toml` using `ini2toml` tool, and the conversion `setup.py` -> `setup.cfg` using `setuptools_py2cfg` tool. You may also want to convert your other projects.

**Comment by chfw** [comment]
could you please resolved the merge conflict?

I could not perform it 

```
git push origin dev
remote: Permission to KOLANICH-libs/pyexcel.git denied to chfw.
fatal: unable to access 'https://github.com/KOLANICH-libs/pyexcel.git/': The requested URL returned error: 403
```

**Comment by KOLANICH** [comment]
I close this PR, feel free to open an own one and push into it.

**Comment by KOLANICH** [comment]
Thanks for the resolving the merge conflicts, BTW.

**Pull request closed without merging**